### PR TITLE
Improve formatting for SE, LV, CZ, SK postcodes + Update tests

### PR DESCRIPTION
**Pull request by Saggre**
### Changes proposed in this Pull Request:

- #### Add data providers for `WC_Formatting_Functions_Test` tests
- #### Add formatting for SE, CZ, SK postcodes. Simplify formatting for LV
- #### Add formatting tests for SE, CZ, SK, DK postcodes

Note: I was unsure whether to include support for formatting international eg. (`CZ-NNN NN`)
 versions of postcodes, but I noticed that it was already supported for danish postcodes, so I added it. If this is something you don't want to support, just tell me and I'll modify the PR.

###### Sweden
https://www.upu.int/UPU/media/upu/PostalEntitiesFiles/addressingUnit/sweEn.pdf
https://en.wikipedia.org/wiki/Postal_codes_in_Sweden
`NNN NN`

###### Czech Republic
https://www.upu.int/UPU/media/upu/PostalEntitiesFiles/addressingUnit/czeEn.pdf
https://en.wikipedia.org/wiki/Postal_codes_in_the_Czech_Republic
`NNN NN` or `CZ-NNN NN`

###### Slovakia
https://www.upu.int/UPU/media/upu/PostalEntitiesFiles/addressingUnit/svkEn.pdf
https://en.wikipedia.org/wiki/Postal_codes_in_Slovakia
`NNN NN `or `SK-NNN NN`

###### Latvia
https://www.upu.int/UPU/media/upu/PostalEntitiesFiles/addressingUnit/lvaEn.pdf
https://en.wikipedia.org/wiki/Postal_codes_in_Latvia
`LV-NNNN`

###### Denmark
https://www.upu.int/UPU/media/upu/PostalEntitiesFiles/addressingUnit/dnkEn.pdf
https://en.wikipedia.org/wiki/Postal_codes_in_Denmark
`NNNN or DK-NNNN`

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run tests `WC_Formatting_Functions_Test`
2. Run legacy tests `WC_Tests_Formatting_Functions`
3. Run tests `WC_Validation_Test`
4. Run legacy tests `WC_Tests_Validation`

### CODE REVIEW

File: plugins/woocommerce/includes/wc-formatting-functions.php [hunk 1]
```diff
@@ -1004,9 +1007,12 @@ function wc_format_postcode( $postcode, $country ) {
 			$postcode = substr_replace( $postcode, ' ', 4, 0 );
 			break;
 		case 'LV':
-			if ( preg_match( '/(?:LV)?-?(\d+)/i', $postcode, $matches ) ) {
-				$postcode = count( $matches ) >= 2 ? "LV-$matches[1]" : $postcode;
-			}
+			$postcode = preg_replace( '/^(LV)?-?(.+)$/', 'LV-${2}', $postcode );
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/wc-formatting-functions.php]: Is there any reason to remove the `(\d+)` here? AFAIK the postcode is all digits, right? So not reason to capture "more than needed". The same applies to CZ/SK above.

**Review by jorgeatorres — CHANGES_REQUESTED** [review]
Hi @Saggre!

Thanks for your contribution 💯. Overall this looks good to me.
Just left some feedback about capturing only digits (not characters in general) in the regexes. Please let me know what you think.

File: plugins/woocommerce/includes/wc-formatting-functions.php (hunk 1 above)
Comment (Saggre) [reply plugins/woocommerce/includes/wc-formatting-functions.php]: You're right, the postcodes are all digits. I think I got the idea for a broader regex from this newly added DK regex:
https://github.com/woocommerce/woocommerce/blob/073c3ff3b420d6cdb608a5b27cdff522a08dc8e6/plugins/woocommerce/includes/wc-formatting-functions.php#L1011-L1013
I narrowed the regex for LV, CZ and SK in a new commit and left DK as is for now.

**Comment by Saggre** [comment]
I made these changes:
- Updated postcode validation for CZ and SK because postcodes including ISO country code prefix and formatted with `wc_format_postcode` did not pass existing validation rules in `WC_Validation::is_postcode`. I think correct and formatted postcodes are expected to automatically pass validation
- Narrowed the regex for LV, CZ and SK postcodes
- Added links to more documentation

**Review by jorgeatorres — APPROVED** [review]
LGTM. Thank you @Saggre!

Just wanted to add that I had to rebase off `trunk` a couple of times to clear CI. I also made a tiny change to one of the regexes.

**Pull request merged**